analyses/integrative-analysis/util/function-samples-integrate.R

```diff
@@ -0,0 +1,211 @@
+############################################################################################################
+#' Function to integrate samples using Seurat
+#' @param seurat_obj_list
+#' @param nfeatures_value
+#' @param num_dim_seurat
+#' @param num_dim_seurat_integration
+#' @param big_data
+#' @param reference_list
+#' @param Genome
+#' @param Regress_Cell_Cycle
+#' @param PCA_Feature_List
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' 
+#'
+seurat_integration <- function(seurat_obj_list, nfeatures_value, num_dim_seurat, num_dim_seurat_integration, 
+                               big_data, reference_list, Genome, Regress_Cell_Cycle, PCA_Feature_List){
+  
+  set.seed(1234) # Make code reproducible
+  
+  seurat_obj_list <- lapply(X = seurat_obj_list, FUN = function(x){
+    
+    cat("Conducting SCTransform for sample:", unique(x$ID), "\n")
+    # the following command replaces NormalizeData(), ScaleData(), and FindVariableFeatures()
+    # Transformed data will be available in the SCT assay, which is set as the default after running sctransform
+    x <- SCTransform(x, variable.features.n = nfeatures_value, verbose = TRUE, return.only.var.genes = FALSE, conserve.memory = TRUE)}) #vst.flavor = "v2"
+  
+  cat("Selecting Intgration Features\n")
+  features <- SelectIntegrationFeatures(object.list = seurat_obj_list, nfeatures = nfeatures_value, verbose = TRUE)
+  seurat_obj_list <- PrepSCTIntegration(object.list = seurat_obj_list, anchor.features = features, verbose = TRUE)
+  seurat_obj_list <- lapply(X = seurat_obj_list, FUN = function(x){
+    x <- RunPCA(x, features = features, verbose = TRUE)})
+  
+  # We will consider projects that contain big data, i.e., >= 1.3M cells.
+  if (big_data == big_data){
```

Comment:
I'm not sure I totally follow what the input value for the big_data param would be? Is it a true/false (or yes/no)? 
Unless I am misunderstanding, I think the condition if(big_data == big_data) will always be true (because big_data would always == itself, right?), so the big_data version of integration would always be run for this function (even if the big_data param was set as FALSE or NO or whatever when the function is called).

Reply:
@sharonfreshour This has been addressed here 3692b112f5cdcd11e36aa53335c1a5c672306dd5.